src/parser/core/modules/Gauge/TimerGauge.ts

```diff
@@ -182,57 +192,58 @@ export class TimerGauge extends AbstractGauge {
 		}
 
 		// Translate state history into a dataset that makes sense for the chart
-		const startTime = this.parser.eventTimeOffset
-		const endTime = startTime + this.parser.pull.duration
+		const endTime = this.parser.pull.timestamp + this.parser.pull.duration
 		const data: ResourceDatum[] = []
+		let previousPaused = false
 		this.history.forEach(entry => {
-			const relativeTimestamp = entry.timestamp - startTime
-
 			// Adjust preceeding data for the start of this state's window
 			const {length} = data
-			if (length > 0 && relativeTimestamp < data[length - 1].time) {
+			const chartY = (this.minimum + entry.remaining) / 1000
+			const prev = data[length - 1]
+			if (length > 0 && entry.timestamp < prev.time) {
 				// If we're updating prior to the previous entry's expiration, update the previous entry
 				// with its state at this point in time - we'll end up with two points showing the update on
 				// this timestamp.
-				const prev = data[length - 1]
-				prev.current = (prev.current || this.minimum / 1000) + ((prev.time - relativeTimestamp) / 1000)
-				prev.time = relativeTimestamp
+				prev.current = (prev.current || this.minimum / 1000) + ((prev.time - entry.timestamp) / 1000)
+				prev.time = entry.timestamp - 1
 
-			} else {
+			} else if (length > 0) {
```

Comment:
I commented a little more on discord about an alternate approach for history tracking that would streamline a lot of this graph generation code, but if we keep the current approach, we don't need this if (length > 0) here - no harm done by adding a 0 at the timestamp of the first generation event.